KeyCloak SSO How to Guide (#1477)
kestra-io/docs: content/docs/15.how-to-guides/keycloak.md

You can follow the steps described [here](https://documentation.cloud-iam.com/get-started/complete-tutorial.html) to deploy a managed KeyCloak cluster for free.

## Configure KeyCloak client

Once in KeyCloak, you would need to create a client:

![alt text](/docs/how-to-guides/client1.png)
![alt text](/docs/how-to-guides/client2.png)

Set `https://{{ yourKestraInstanceURL }}/oauth/callback/keycloak` as Valid redirect URIs and `https://{{ yourKestraInstanceURL }}/logout` as Valid post logout redirect URIs.

![alt text](/docs/how-to-guides/redirect-uri.png)

## Kestra Configuration

```yaml
micronaut:
  security:
    oauth2:
      enabled: true
      clients:
        keycloak:
          client-id: "{{ clientId }}"
          client-secret: "{{ clientSecret}}"
          openid:
            issuer: "https://{{ keyCloakServer }}/auth/realms/{{yourRealm}}"
    endpoints:
      logout:
        get-allowed: true
```

You can retrieve `clientId` and `clientSecret` via KeyCloak user interface

![alt text](/docs/how-to-guides/clientId.png)
![alt text](/docs/how-to-guides/clientSecret.png)

> Don't forget to set a default role in your Kestra configuration to allow new users to connect properly.

Reviewer: ```suggestion
Don't forget to set a default role in your Kestra configuration to streamline the process of adding new users.
```
